File: php/features.php
<?php
declare(strict_types=1);

// AntiPhishingDetection SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AntiPhishingDetectionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AntiPhishingDetectionBaseFeature();
            case "test":
                return new AntiPhishingDetectionTestFeature();
            default:
                return new AntiPhishingDetectionBaseFeature();
        }
    }
}

File: py/features.py
# AntiPhishingDetection SDK feature factory

from feature.base_feature import AntiPhishingDetectionBaseFeature
from feature.test_feature import AntiPhishingDetectionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AntiPhishingDetectionBaseFeature(),
        "test": lambda: AntiPhishingDetectionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
